src/hooks/mixpanel.tsx

```diff
@@ -0,0 +1,49 @@
+import mixpanel from 'mixpanel-browser'
+import React, { FC, ReactNode, createContext, useCallback, useContext, useEffect, useMemo, useState } from 'react'
+
+interface MixPanelProviderProps {
+  children: ReactNode
+  mixpanelToken?: string
+}
+
+type MixPanelContextType = {
+  track: (event: string, properties: { [x: string]: any }) => void
+}
+
+/* eslint-disable @typescript-eslint/no-unused-vars */
+const dummyTrack = (event: string, properties: { [x: string]: any }) => {
+  // This is intentional
+}
+
+export const MixPanelContext = createContext<MixPanelContextType>({ track: dummyTrack })
+
+export const MixPanelProvider: FC<MixPanelProviderProps> = ({ children, mixpanelToken }: MixPanelProviderProps) => {
+  const [activedMixPanel, setActivedMixPanel] = useState(false)
+
+  useEffect(() => {
+    if (mixpanelToken) {
+      try {
+        mixpanel.init(mixpanelToken)
+        setActivedMixPanel(true)
+      } catch (error) {
+        console.error('Error activating Mixpanel: ', error)
+      }
+    }
+  }, [])
+
+  const track = useCallback((event: string, properties: { [x: string]: any }) => {
+    mixpanel.track(event, { ...properties, source: 'interface' })
```

Comment:
in **components** lets change this order to 
```js
mixpanel.track(event, { source: 'pangolin-components', ...properties,  });
```

so that in **interface** we can use component implementation by overriding source to "interface"

```ts
export function useMixpanel() {
  const { track } = useComponentMixpanel();

  const overrideTrack = useCallback((event: MixPanelEvents, properties: { [x: string]: any }) => {
    track(event, { source: 'interface', ...properties });
  }, [track]);

  return { track: overrideTrack };
}
``` 

in that way we don't need to define entire mixpanel implementation here. Just need to override useMixPanel like above